src/ucp/core/ucp_listener.c

```diff
@@ -71,12 +85,26 @@ static ucs_status_t ucp_listener_conn_request_callback(void *arg,
         goto err;
     }
 
-    /* create endpoint to the worker address we got in the private data */
-    status = ucp_ep_create_to_worker_addr(listener->wiface.worker, &params,
-                                          &remote_address, UCP_EP_CREATE_AM_LANE,
-                                          "listener", &ep);
-    if (status != UCS_OK) {
-        goto err_free_address;
+    if (client_data->is_full_addr) {
+        /* create endpoint to the worker address we got in the private data */
+        status = ucp_ep_create_to_worker_addr(listener->wiface.worker, &params,
+                                              &remote_address, UCP_EP_CREATE_AM_LANE,
+                                              "listener", &ep);
+        if (status != UCS_OK) {
+            goto err_free_address;
+        }
+
+        ep->flags &= ~UCP_EP_FLAG_LISTENER;
```

Comment:
why need to clear this flag? do we expect it to be set before?

Reply:
no, will remove this.